src/Common/Api/Parameter.php

```diff
@@ -285,6 +285,11 @@ private function hasCorrectType($userValue): bool
             return true;
         }
 
+        // allow string nulls
+        if ($this->type == 'string' && $userValue === null) {
```

Comment:
It's mentioned in the commit message for: https://github.com/php-opencloud/openstack/commit/f377fa58383f8c100495504b890b3e34f0688be0

>This is needed because for the subnet::update() call gatewayIp can be 'null' even though it's a string type.  Null in this case means "remove the gatewayIp".